compiler/qsc_linter/src/linter.rs

```diff
@@ -211,11 +254,21 @@ impl Display for LintLevel {
     }
 }
 
-/// End-user configuration for each lint level.
+/// End-user configuration for a specific lint or a lint group.
+#[derive(Debug, Clone, Deserialize, Serialize, PartialEq, Eq)]
+#[serde(untagged)]
+pub enum LintOrGroupConfig {
```

Comment:
Since `level` is a common field, I probably would have organized the types like this - fewer structs. It doesn't make a meaningful difference at runtime, and it might make deserialization awkward I don't know. So take it or leave it - I just wanted to make sure you considered it

```rs
struct LintConfig {
    kind: LintConfigKind,
    level: LintLevel,
}

enum LintConfigKind {
    Lint(LintKind),
    Group(LintGroup)
}
```

Reply:
Hmm, I just saw `unfold_group` above - my suggestion would probably make that function more awkward. So I can see why you did it this way